### PR TITLE
Support squeeze in bert model

### PR DESCRIPTION
**Pull request by mcw-zwakeelTT**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/12851

### Problem description
ttnn squeeze is only supported for dim = 0. So for such inputs convert aten squeeze to ttnn squeeze.
While for dim != 0 inputs, converted aten squeeze to ttnn reshape keeping input in row major layout.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -98,7 +98,8 @@ def compile_and_run(device, reset_torch_dynamo, request):
             outputs_after = run_model(m, inputs)
             end = time.perf_counter() * 1000
             comp_runtime_metrics = {"success": True, "run_time": round(end - start, 2)}
-            option._out_fx_graphs[0].print_tabular()
+            if option.verbose:
+                option._out_fx_graphs[0].print_tabular()
```
Comment (kevinwuTT) [inline tests/conftest.py:102]: This change doesn't seem relevant to this particular commit.

File: torch_ttnn/backend.py [hunk 2]
```diff
@@ -189,12 +189,13 @@ def aten_backend(
...
-            print(f"{node.name}: {node.meta}")
+    if option.verbose:
+        for node in gm.graph.nodes:
+            if node.op == "placeholder":
+                print(f"{node.name}: {node.meta}")
 
-    for number, line in enumerate(gm.code.splitlines()):
-        print(f"{number + 1}: {line}")
+        for number, line in enumerate(gm.code.splitlines()):
+            print(f"{number + 1}: {line}")
```
Comment (kevinwuTT) [inline torch_ttnn/backend.py:198]: Same here. These changes don't seem relevant to this commit.

File: tests/conftest.py (hunk 1 above)
Comment (mcw-zwakeelTT) [reply tests/conftest.py:102]: Yeah not relevant to this specific PR, but printing should be optional depending on verbosity.

Comment (ayerofieiev-tt) [reply tests/conftest.py:102]: Not relevant change has to be moved out to a separate PR. I know its annoying, but thats the eng culture to project. Please do so.

I will do better myself too.

File: torch_ttnn/passes/lowering/add_data_move_pass.py [hunk 3]
```diff
@@ -324,8 +324,10 @@ def try_add_layout_change_after_node(src_node, dst_idx, dst_node) -> torch.fx.no
     g = dst_node.graph
     with g.inserting_before(dst_node):
         to_layout = g.call_function(ttnn.to_layout, (dst_node, TtnnTileLayout()))
+    with g.inserting_before(dst_node):
+        to_device = g.call_function(ttnn.to_device, (to_layout, TtnnDevice()))
```
Comment (ayerofieiev-tt) [inline torch_ttnn/passes/lowering/add_data_move_pass.py:328]: why? we have a separate pass moving to device, no?

File: torch_ttnn/passes/lowering/to_tt_pass.py [hunk 4]
```diff
@@ -594,6 +588,14 @@ def rewrite_node(node):
                     new_nodes.append(g.call_function(ttnn.reshape, args=(new_nodes[-1], output_size)))
                 return new_nodes[-1]
 
+            if node.target == torch.ops.aten.squeeze.dim:
```
Comment (ayerofieiev-tt) [inline torch_ttnn/passes/lowering/to_tt_pass.py:591]: Please remove, ttnn.squeeze supports dim != 0 now and internally calls reshape.
